spec/controllers/users_controller_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+describe UsersController, type: :controller do
+
+  describe 'User creation' do
+    let(:admin) {create(:user, :admin)}
+    let(:hacker) {create(:user)}
+
+    context 'logged in as an admin' do
+      before :each do
+        allow(controller).to receive(:current_user) { admin }
+      end
+
+      it 'is able to create a user' do
+        expect{
+          post :create, user: attributes_for(:user)
+        }.to change(User, :count).by(2)
```

Comment:
Good question! I have no friggan idea. My only guess is that it's counting the admin account? I'm doing something wrong.

> On Sep 13, 2015, at 8:04 PM, Shaun Butler <notifications@github.com> wrote:
> 
> In spec/controllers/users_controller_spec.rb:
> 
> > +
> > +describe UsersController, type: :controller do
> > +
> > +  describe 'User creation' do
> > +    let(:admin) {create(:user, :admin)}
> > +    let(:hacker) {create(:user)}
> > +
> > +    context 'logged in as an admin' do
> > +      before :each do
> > +        allow(controller).to receive(:current_user) { admin }
> > +      end
> > +
> > +      it 'is able to create a user' do
> > +        expect{
> > +          post :create, user: attributes_for(:user)
> > +        }.to change(User, :count).by(2)
> why 2??
> 
> —
> Reply to this email directly or view it on GitHub.
>